contracts/contango/ContangoCauldron.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/utils-v2/contracts/math/WMul.sol";
+import "@yield-protocol/utils-v2/contracts/math/WDivUp.sol";
+import "../constants/Constants.sol";
+import "../Cauldron.sol";
+import "./util/EnumerableSet.sol";
+
+contract ContangoCauldron is Cauldron {
+    using CauldronMath for uint128;
+    using EnumerableSet for EnumerableSet.Bytes6Set;
+    using CastU256I256 for uint256;
+    using WMul for uint256;
+    using WDivUp for uint256;
+
+    EnumerableSet.Bytes6Set private assetsInUse;
+    mapping(bytes6 => DataTypes.Balances) public balancesPerAsset;
+    int256 public peekFreeCollateralUSD;
+
+    uint128 public collateralisationRatio;
+    bytes6 public commonCcy; // Currency to use as common ground for all the ink & art
+
+    constructor(
+        uint128 _collateralisationRatio,
+        bytes6 _commonCurrency,
+        uint8 _commonCcyDecimals
+    ) {
+        collateralisationRatio = _collateralisationRatio * uint128(10**(18 - _commonCcyDecimals));
```

Comment:
Why not just passing the ratio?

Reply:
no reason, started as storing the 3 things and ended up like this, will change it